### ISSUE TITLE
Notification is blocking user interaction in dialog

### ISSUE THREAD

**Issue by mukherjeesudebi** [vaadin/flow-components#6905]
### Description

Notification.show is blocking user interaction in dialogs. In some cases it also closes a dialog when interacting with a button click.
**The issue is replicable in 24.5.6 and 24.5.7 but it works fine in 24.5.5**

### Expected outcome

Notification.show should work as expected and should not interfere with user interaction in dialogs or close dialogs with user clicks inside the dialog.

### Minimal reproducible example

```
@Route("")
public class MainView extends VerticalLayout {
	public MainView() {
		TextField mainViewTextField = new TextField("Main View Time");
		mainViewTextField.setValue("Click Button to Set Main View Time");
		mainViewTextField.setWidth("100%");

		Button mainViewSetTimeButton = new Button("Set Time", e -> {
			mainViewTextField.setValue("Time: " + System.currentTimeMillis());
		});

		Button mainViewShowNotificationButton = new Button("Main View Notification", e -> {
			Notification.show("MainView Notification " + System.currentTimeMillis());
		});

		Button mainViewOpenDialog1Button = new Button("Open Dialog 1", mainViewE -> {
			Dialog dialog1 = new Dialog();
			dialog1.setWidth("80%");
			dialog1.setCloseOnOutsideClick(true);

			TextField dialog1TextField = new TextField("Dialog 1 Time");
			dialog1TextField.setValue("Click Button to Set Dialog 1 Time");
			dialog1TextField.setWidth("100%");

			Button dialog1SetTimeButton = new Button("Set Time", e -> {
				dialog1TextField.setValue("Time: " + System.currentTimeMillis());
			});

			Button dialog1ShowNotificationButton = new Button("Dialog 1 Notification", e -> {
				Notification.show("Dialog 1 Notification " + System.currentTimeMillis());
			});

			Button dialog1CloseButton = new Button("Dialog 1 Close", e -> {
				dialog1.close();
			});

			Button dialog1ShowNotificationAndCloseButton = new Button("Dialog 1 Notification and Close", e -> {
				Notification.show("Dialog 1 Notification " + System.currentTimeMillis());
				dialog1.close();
			});

			Button dialog1OpenDialog2Button = new Button("Open Dialog 2", dialog1E -> {
				Dialog dialog2 = new Dialog();
				dialog2.setWidth("80%");
				dialog2.setCloseOnOutsideClick(true);

				TextField dialog2TextField = new TextField("Dialog 2 Time");
				dialog2TextField.setValue("Click Button to Set Dialog 2 Time");
				dialog2TextField.setWidth("100%");

				Button dialog2SetTimeButton = new Button("Set Time", e -> {
					dialog2TextField.setValue("Time: " + System.currentTimeMillis());
				});

				Button dialog2ShowNotificationButton = new Button("Dialog 2 Notification", e -> {
					Notification.show("Dialog 2 Notification " + System.currentTimeMillis());
				});

				Button dialog2CloseButton = new Button("Dialog 2 Close", e -> {
					dialog2.close();
				});

				Button dialog2ShowNotificationAndCloseButton = new Button("Dialog 2 Notification And Close", e -> {
					Notification.show("Dialog 2 Notification " + System.currentTimeMillis());
					dialog2.close();
				});

				dialog2.add(new VerticalLayout(dialog2TextField, dialog2SetTimeButton, dialog2ShowNotificationButton,
						dialog2CloseButton, dialog2ShowNotificationAndCloseButton));
				dialog2.open();
			});

			dialog1.add(new VerticalLayout(dialog1TextField, dialog1SetTimeButton, dialog1ShowNotificationButton,
					dialog1CloseButton, dialog1OpenDialog2Button, dialog1ShowNotificationAndCloseButton));
			dialog1.open();
		});

		add(mainViewTextField, mainViewSetTimeButton, mainViewShowNotificationButton, mainViewOpenDialog1Button);
	}
}
```

### Steps to reproduce

1. Add the code to a vaadin 24.5.7 starter sample
2. Test cases are as follows : 

**Case 1**
Dialog 2 has a click button Dialog 2 Notification and Close that shows a notification and closes itself (closes Dialog 2). Dialog 2 closes but the user cannot interact with the application until the notification disappears. Once the notification disappears, any user click closes Dialog 1 instead of the user being able to interact with Dialog 1. There are 2 bugs in this scenario.

**Test Case**

- Click Open Dialog 1 Button
- Click Set Time Button several times to see that it works as expected
- Click Open Dialog 2 Button
- Click Dialog 2 Notification and Close Button
- Try to Click Set Time Button on Dialog 1 before the Notification closes
 **BUG:** **Nothing is happening**
- Click on Set Time Button on Dialog 1 after the Notification closes
 **BUG:** **Instead of Setting the time, Dialog 1 closes**

**Case 2**
The same thing happens if we try to open and close the dialog when notification from Dialog1 is still visible

**Test Case**

- Click Open Dialog 1 Button
- Click Set Time Button several times to see that it works as expected
- Click Dialog 1 Notification Button
- Click Open Dialog 2 Button before the notification from dialog 1 disappears
- Click Dialog 2 Close Button before the notification from dialog 1 disappears
- Try to Click Set Time Button on Dialog 1 before the Notification closes
 **BUG: Nothing is happening**
- Click on Set Time Button on Dialog 1 after the Notification closes
 **BUG: Instead of Setting the time, Dialog 1 closes**

**Case 3**
Dialog 2 has a click button Dialog 2 Notification that shows a notification. If the user closes Dialog 2 by using the Dialog 2 Close button before the notification disappears, any interaction with Dialog 1 will cause Dialog 1 to close. 

**Test Case**

- Click Open Dialog 1 Button
- Click Set Time Button several times to see that it works as expected
- Click Open Dialog 2 Button
- Click Dialog 2 Notification Button
- Click Dialog 2 Close Button before the notification disappears
- Click Set Time Button on Dialog 1 
 **BUG: Instead of Setting the time, Dialog 1 closes.**

**Case 4**
Dialog 1 and Dialog 2 can be closed when clicking outside the dialog or pressing the escape key. When there is a Notification.show, the dialog cannot be closed by clicking outside the dialog or pressing the escape key until the notification disappears. 

**Test Case**

- Click Open Dialog 1 Button
- Click outside the dialog or press the escape key to test closing the dialog
- Repeat steps 1 and 2 until satisfied that it works as expected
- Click Dialog 1 Notification Button
- Attempt to close the dialog by pressing the escape key or clicking outside the dialog before the notification disappears.
 **BUG: The dialog will not close as long as the notification shows.**

### Environment

Vaadin version(s): 24.5.6 and 24.5.7. Works fine in 24.5.5

**Comment by web-padawan** [comment]
This is already fixed by https://github.com/vaadin/web-components/pull/8291 which will be included in the next 24.5 patch release.

**Comment by mukherjeesudebi** [comment]
Thanks for the update
